devDocs/startupShutdown.md

```diff
@@ -0,0 +1,173 @@
+ - NVDA installed
+ - Enable "Start NVDA after I sign in"
+
+Steps:
+ 1. Start Windows
+ 1. Sign in
+ 1. Observe NVDA starts
+
+### Running the *.exe
+
+Steps:
+ 1. Press `win+r`
+ 1. Enter <path to nvda.exe>
+ 1. Press enter
+ 1. Observe NVDA starts
+
+Variation:
+- using an installer .exe, an installed copy .exe, or a portable copy .exe
+   - the installer allows you to create an installed copy and a portable copy
+- CLI options
+
+### Running from source (runnvda.bat)
+Prerequisites
+- clone project and build NVDA (see [project readme](https://github.com/nvaccess/nvda/blob/master/readme.md#getting-the-source-code)).
+
+Steps:
+ 1. Run `runnvda.bat` from cmd
+ 1. Observe NVDA starts
+
+### An input gesture to restart
+
+Prerequisite:
+- Input gesture to restart is assigned to a key
+
+Steps:
+ 1. Press (or emulate) the input gesture
+ 1. Observe that NVDA exits
+ 1. Observe that a new instance is started
+
+## Technical notes
+
+These notes are aimed at developers, wishing to understand technical aspects of the NVDA start and exit.
+
+1. No more than one NVDA process instance should be running time. Interactions with itself could cause severe issues, some (non-exhaustive list) examples of sub-systems where this would be a problem:
+   - NVDA config files
+   - Global (OS level) keyboard hook
+   - Changed / incompatible in-process code
+2. As such, we want to be able to detect running instances, cause them to exit, and confirm they have exited.
+
+### Exit hooks/triggers
+
+There are 3 ways that NVDA receives a request to exit:
+
+- From internally calling [triggerNVDAExit](#When-exiting-from-triggerNVDAExit)
+- Receiving [WM_QUIT](#When-exiting-from-WM_QUIT) Windows message
+- Receiving [wx.EVT_END_SESSION](#When-exiting-from-wxEVT_END_SESSION) due to Windows session ending
+
+### When exiting from `triggerNVDAExit`
+* Called from within NVDA.
+* A function in the core module
+* Only executes the code once, uses a lock and flag to ensure this
+* Uses a queue on the main thread to queue a safe shutdown
+* Once the queued shutdown starts:
+    1. the updateCheck is terminated
+    1. watchdog is terminated
+    1. addons and the brailleViewer are terminated, so we can close all windows safely
+    1. All wx windows are closed
+    1. Now that windows are closed, a new NVDA instance is started if requested
+
+### When exiting from `WM_QUIT`
+* [A Windows Message](https://docs.microsoft.com/en-us/windows/win32/winmsg/wm-quit) received from an external process, such as another NVDA process.
+* NVDA accepts `WM_QUIT` messages from other processes and creates a [named window](https://docs.microsoft.com/en-us/windows/win32/learnwin32/creating-a-window#creating-the-window) that can be discovered.
+* `WM_QUIT` is handled by `wx`, which force closes all wx windows (other UI features like the systray icon are not windows, and remain) and then exits the main loop. `triggerNVDAExit` is a more expansive check than how wxWidgets handles `WM_QUIT````

```

Comment:
There are too many grave characters on the end of WM_QUIT

Reply:
```suggestion
* `WM_QUIT` is handled by `wx`, which force closes all wx windows (other UI features like the systray icon are not windows, and remain) and then exits the main loop.
`triggerNVDAExit` is a more expansive check than how wxWidgets handles `WM_QUIT`
```